tests/framework/ROM/TimeDepSKL/projectile.py

```diff
@@ -0,0 +1,129 @@
+#***************************************
+#
+# Simulates time-dependent track of a projectile through the air from start to 0,
+#     assuming no air resistance.
+#     Inputs:
+#       (x0,y0) - initial position
+#       v0 - initial total velocity
+#       ang - angle of initial motion, in degrees, with respect to flat ground
+#     Outputs:
+#       (x,y) - vector positions of projectile in time
+#       t - corresponding time steps
+#
+import numpy as np
+
+def prange(v,th,y0=0,g=9.8):
+  """
+    Calculates the analytic range.
+    @ In, v, float, velocity of the projectile
+    @ In, th, float, angle to the ground for initial projectile motion
+    @ In, y0, float, optional, initial height of projectile
+    @ In, g, float, optional, gravitational constant (m/s/s)
+    @ Out, prange, float, range
+  """
+  return v*np.cos(th)/g * (v*np.sin(th) + np.sqrt(v*v*np.sin(th)**2+2.*g*y0))
+
+def time_to_ground(v,th,y0=0,g=9.8):
+  """
+    Calculates the analytic time of flight
+    @ In, v, float, velocity of the projectile
+    @ In, th, float, angle to the ground for initial projectile motion
+    @ In, y0, float, optional, initial height of projectile
+    @ In, g, float, optional, gravitational constant (m/s/s)
+    @ Out, time_to_ground, float, time projectile is above the ground
+  """
+  return v*np.sin(th)/g + np.sqrt(v*v*np.sin(th)**2+2.*g*y0)/g
+
+def x_pos(x0,v,t):
+  """
+    Calculates the x position in time
+    @ In, x0, float, initial horizontal position
+    @ In, v, float, velocity of the projectile
+    @ In, t, float, time of flight
+    @ Out, x_pos, float, horizontal position
+  """
+  return x0 + v*t
+
+def y_pos(y0,v,t):
+  """
+    Calculates the analytic vertical position in time
+    @ In, y0, float, initial vertical position
+    @ In, v, float, velocity of the projectile
+    @ In, t, float, time of flight
+    @ Out, y_pos, float, vertical position
+  """
+  return y0 + v*t - 4.9*t*t
+
+def run(self,Input):
+  """
+    Method require by RAVEN to run this as an external model.
+    @ In, self, object, object to store members on
+    @ In, Input, dict, dictionary containing inputs from RAVEN
+    @ Out, None
+  """
+  x0 = Input.get('x0',0.0)
+  y0 = Input.get('y0',0.0)
+  v0 = Input.get('v0',1.0)
+  ang = Input.get('angle',45.)*np.pi/180.
+  self.x0 = x0
+  self.y0 = y0
+  self.v0 = v0
+  self.ang = ang
+
+  ts = np.linspace(0,0.1,10)
+  #ts = np.linspace(0,time_to_ground(v0,ang,y0),20)
+
+  vx0 = np.cos(ang)*v0
+  vy0 = np.sin(ang)*v0
+  r = prange(v0,ang,y0)
+
+  self.x = np.zeros(len(ts))
+  self.y = np.zeros(len(ts))
+  self.r = np.zeros(len(ts))
+  for i,t in enumerate(ts):
+    self.x[i] = x_pos(x0,vx0,t)
+    self.y[i] = y_pos(y0,vy0,t)
+    self.r[i] = r
+  self.time = ts
+
+#can be used as a code as well
+if __name__=="__main__":
+  import sys
+  inFile = sys.argv[sys.argv.index('-i')+1]
+  outFile = sys.argv[sys.argv.index('-o')+1]
+  #construct the input
+  Input = {}
```

Comment:
I think we need to change the name here.

Reply:
I disagree, I think developers have autonomy within their external models.